articat/fs_artifact.py

```diff
@@ -73,9 +73,12 @@ def main_dir(self) -> str:
         """
         # to not worry about presence of scheme, we remove the prefix
         # and add it back
-        if self.files_dir:
+        if self._file_prefix is not None:
+            return self._file_prefix
+        elif self.files_dir:
```

Comment:
So `_file_prefix` is short for `_staging_file_prefix`, e.g. it's only set for staged artifacts? And `files_dir` is `main_dir`, but only for published artifacts?

Something to think about for the future... should the whole FSArtifact be refactored such that:

- main_dir (or files_dir... either name okay) holds the path to the staged artifact and then the published artifact
- a bool variable like `is_staging` indicates whether the artifact is still unpublished
- files_pattern is relative to `main_dir` and therefore doesn't need to be edited upon publication.

Reply:
> So _file_prefix is short for _staging_file_prefix, e.g. it's only set for staged artifacts? And files_dir is main_dir, but only for published artifacts?

Yep.

Re refactor/future - def agree, there's a number of things I would like to refactor or add, but that probably won't happen anytime soon, not because I don't want to, but because it's not necessarily our priority right now. Linking to #5